File: ProjectEuler/Problems 1-33/Power digit sum.cpp

#include <iostream>
#include <math.h>
#include <iomanip>
using namespace std;

///////////////////// Resorted to java for this

int main()
{
	double num= pow(2, 1000);
	//double num = 235;
	printf("%f", num);
	
	double added = 0;
	
	while(num > 0)
	{
		added += (unsigned long  long int) num % 10;
		num /= 10;
	}
	cout << endl << added;
	
}

File: ProjectEuler/Problems 1-33/PowerDigitSum.java

import java.math.BigInteger;
import static java.lang.Math.pow;

public class PowerDigitSum
{
	public static void main(String[] args)
	{
		int added = 0;
		int power = 1000;

		BigInteger num = new BigInteger("2");
		BigInteger sum = new BigInteger("0");
		//num2 = num.pow(power);

		num = num.pow(1000);

		while(num.signum() == 1)
		{
			sum = sum.add(num.mod(BigInteger.TEN));
			num = num.divide(BigInteger.TEN);
		}
		System.out.println(sum);

	}

}